pkgs/development/compilers/koka/default.nix

```diff
@@ -0,0 +1,49 @@
+{ stdenv, cmake, mkDerivation, fetchFromGitHub, alex, array, base, bytestring, cond
+, containers, directory, extra, filepath, haskeline, hpack, hspec
+, hspec-core, json, lib, mtl, parsec, process, regex-compat, text, time }:
+
+let
+  version = "2.1.1";
+  src = fetchFromGitHub {
+    owner = "koka-lang";
+    repo = "koka";
+    rev = "v${version}";
+    sha256 = "sha256-cq+dljfTKJh5NgwQfxQQP9jRcg2PQxxBVEgQ59ll36o=";
+    fetchSubmodules = true;
+  };
+  kklib = stdenv.mkDerivation {
+    pname = "kklib";
+    inherit version;
+    src = "${src}/kklib";
+    nativeBuildInputs = [ cmake ];
+  };
+in
+mkDerivation rec {
+  pname = "koka";
+  inherit version src;
+  isLibrary = false;
+  isExecutable = true;
+  libraryToolDepends = [ hpack ];
+  executableHaskellDepends = [
+    array base bytestring cond containers directory haskeline mtl
+    parsec process text time cmake kklib
+  ];
+  executableToolDepends = [ alex ];
+  testHaskellDepends = [
+    array base bytestring cond containers directory extra filepath
+    haskeline hspec hspec-core json mtl parsec process regex-compat
+    text time
+  ];
+  postInstall = ''
```

Comment:
You should probably wrap the executable here as well with its run time dependencies. I think we need at least `cmake`, `gcc` and related binutils. 

good way to test is to try and use `koka` inside a `nix-shell --pure -p koka -I nixpkgs=/path/to/nixpkgs`

Reply:
Also be mindful that `stdenv` contains a C compiler in the nix-shell by default.